### PR TITLE
test: add ssr hydration tests

### PR DESCRIPTION
**Pull request by pgmanutd**
This test ensures that ssr hydration works correctly with compiled.

### CODE REVIEW

File: packages/core/src/__tests__/server-side-hydrate.test.tsx [hunk 1]
```diff
@@ -0,0 +1,37 @@
+import React from 'react';
+import { renderToString } from 'react-dom/server';
+import ReactDOM from 'react-dom';
+import { styled } from '@compiled/core';
+
+describe('server side hydrate', () => {
+  it('should ssr then hydrate element correctly', () => {
```
Comment (madou) [inline packages/core/src/__tests__/server-side-hydrate.test.tsx:7]: could we rename this to `it should not log unexpected warnings when hydrating SSRing`

File: packages/core/src/__tests__/server-side-hydrate.test.tsx [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    delete process.release.name;
+
+    ReactDOM.hydrate(element, elem);
+
+    // Restore the release name after hydration is complete
+    process.release.name = releaseName;
+
+    const mockCalls = (console.error as jest.Mock).mock.calls.filter(
+      ([f, s]) =>
+        !(f === 'Warning: Did not expect server HTML to contain a <%s> in <%s>.%s' && s === 'style')
```
Comment (madou) [inline packages/core/src/__tests__/server-side-hydrate.test.tsx:32]: can we add a comment here explaining that we will get mismatches for the style rendering but this is expected

File: packages/core/src/__tests__/server-side-hydrate.test.tsx [hunk 3]
```diff
@@ -4,7 +4,7 @@ import ReactDOM from 'react-dom';
 import { styled } from '@compiled/core';
 
 describe('server side hydrate', () => {
-  it('should ssr then hydrate element correctly', () => {
+  it('it should not log unexpected warnings when hydrating SSRing', () => {
```
Comment (madou) [inline packages/core/src/__tests__/server-side-hydrate.test.tsx:7]: remove the it :P

Comment (pgmanutd) [reply packages/core/src/__tests__/server-side-hydrate.test.tsx:7]: fuck

**Review by madou — APPROVED** [review]

**Pull request merged**